Derivs/src/derivs_spacetimex.hxx

```diff
@@ -0,0 +1,512 @@
+#ifndef CARPETX_DERIVS_DERIVS_HXX
+#define CARPETX_DERIVS_DERIVS_HXX
+
+#include <loop_device.hxx>
+#include <mat.hxx>
+#include <simd.hxx>
+#include <vec.hxx>
+
+#include <cctk.h>
+#include <cctk_Arguments.h>
+#include <cctk_Parameters.h>
+
+#include <type_traits>
+// Unified derivative header
+
+// TODO: Make this a runtime parameter
+constexpr int deriv_order = 4;
+
+////////////////////////////////////////////////////////////////////////////////
+
+using Arith::pow2;
+
+template <typename T>
+inline ARITH_INLINE ARITH_DEVICE ARITH_HOST T pow3(const T &x) {
+  return pow2(x) * x;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+
+using Arith::simd;
+using Arith::simdl;
+
+template <typename T>
+inline ARITH_INLINE ARITH_DEVICE ARITH_HOST simd<T>
+deriv1d(const simdl<T> &mask, const T *restrict const var, const ptrdiff_t di,
+        const T dx) {
+  const auto load = [&](const int n) {
+    return maskz_loadu(mask, &var[n * di]) - maskz_loadu(mask, &var[-n * di]);
+  };
+  if constexpr (deriv_order == 2)
+    return 1 / T(2) * load(1) / dx;
+  if constexpr (deriv_order == 4)
+    return (-1 / T(12) * load(2) + 2 / T(3) * load(1)) / dx;
+  if constexpr (deriv_order == 6)
+    return (1 / T(60) * load(3) - 3 / T(20) * load(2) + 3 / T(4) * load(1)) /
+           dx;
+}
+
+using Loop::GF3D2;
```

Comment:
It's weird to sprinkle the `using` declarations across the file. They should either be inside functions, or should be collected near the top similar to `#include` statements.